Quickstarts : Update Micronaut version in `quickstarts/gradle/micronaut`

### Component

None

### Task description

### Description

This is a follow-up to #2753

Our Micronaut Gradle quickstarts are using an outdated version of [Micronaut framework](https://micronaut.io/):
https://github.com/eclipse/jkube/blob/b64915ec6090dcb0171e45b6372052278210e6f3/quickstarts/gradle/micronaut-customized-image/gradle.properties#L15
https://github.com/eclipse/jkube/blob/b64915ec6090dcb0171e45b6372052278210e6f3/quickstarts/gradle/micronaut/gradle.properties#L15

Micronaut 4 now requires Java 17 as baseline. Maybe we should not update this quickstart directly to 4.x. For now, updating it to the latest 3.x version (`3.9.7` at the time of creating this issue) would be okay.

:warning: This requires setting up some local Kubernetes Cluster ([MiniKube](https://minikube.sigs.k8s.io/docs/start/)/[Kind](https://kind.sigs.k8s.io/))

After updating Micronaut version, you should verify that quickstart is building fine and is also working as specified in their respective [README](https://github.com/eclipse/jkube/blob/master/quickstarts/maven/micronaut-customized-image/README.md) . You can take a look at this YouTube video [Sprint 192 - Eclipse JKube Micronaut integration](https://youtu.be/I3pjd0D90gA) to see how to try it out.

### Expected Behavior
Micronaut Quickstart is updated to latest `3.x` version

### Acceptance Criteria
- [x] Micronaut Quickstarts are updated to latest `3.x` version
- [ ] Micronaut Quickstarts work as documented in their respective README.md files

### Before you start :red_circle:


:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down::point_down::point_down::point_down::point_down::point_down::point_down:
Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.


### How to _manually_ test my changes


#### Kubernetes


If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.


- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)


#### OpenShift


If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.


Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

Hi @manusa ,

Can you help me with this issue, unable to build micronaut-customized-image:



Downloaded from central: https://repo.maven.apache.org/maven2/com/marcnuri/helm-java/linux-amd64/0.0.6/linux-amd64-0.0.6.jar (24 MB at 7.1 MB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.508 s
[INFO] Finished at: 2024-04-17T01:19:47Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.16.2:build (default-cli) on project micronaut-customized-image: Execution default-cli of goal org.eclipse.jkube:kubernetes-maven-plugin:1.16.2:build failed: No <dockerHost> given, no DOCKER_HOST environment variable, no read/writable '/var/run/docker.sock' or '//./pipe/docker_engine' and no external provider like Docker machine configured -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
micronaut-customized-image (master) $

--- a/quickstarts/gradle/micronaut-customized-image/build.gradle
+++ b/quickstarts/gradle/micronaut-customized-image/build.gradle
@@ -15,6 +15,7 @@ plugins {
     id("com.github.johnrengelman.shadow") version "7.0.0"
     id("io.micronaut.application") version "1.5.4"
     id("org.eclipse.jkube.kubernetes") version "1.17.0"
+    id("org.eclipse.jkube.openshift") version "1.17.0"
 }
 
 version = "1.5.1"
@@ -40,6 +41,7 @@ dependencies {
     implementation("javax.annotation:javax.annotation-api")
     runtimeOnly("ch.qos.logback:logback-classic")
     implementation("io.micronaut:micronaut-validation")
+    implementation("io.micronaut.rxjava2:micronaut-rxjava2")
 }
 
 application {

--- a/quickstarts/gradle/micronaut-customized-image/gradle.properties
+++ b/quickstarts/gradle/micronaut-customized-image/gradle.properties
@@ -12,4 +12,4 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-micronautVersion=2.5.11
+micronautVersion=3.9.7

--- a/quickstarts/gradle/micronaut/gradle.properties
+++ b/quickstarts/gradle/micronaut/gradle.properties
@@ -12,7 +12,7 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-micronautVersion=3.2.3
+micronautVersion=3.9.7
 org.gradle.java.installations.auto-download=false
 org.gradle.java.installations.auto-detect=false
 org.gradle.java.installations.fromEnv=JAVA_HOME

